src/plugins/workspace/server/saved_objects/workspace_id_consumer_wrapper.ts

```diff
@@ -48,6 +52,32 @@ export class WorkspaceIdConsumerWrapper {
     return type === UI_SETTINGS_SAVED_OBJECTS_TYPE;
   }
 
+  private validateObjectInAWorkspace<T>(
+    object: SavedObject<T>,
+    workspace: string,
+    request: OpenSearchDashboardsRequest
+  ) {
+    // Data source is a workspace level object, validate if the request has access to the data source within the requested workspace.
+    if (validateIsWorkspaceDataSourceAndConnectionObjectType(object.type)) {
+      if (!!getWorkspaceState(request).isDataSourceAdmin) {
+        return true;
+      }
+      // Deny access if the object is a global data source (no workspaces assigned)
```

Comment:
Seems the dashboards admin will be denied too, is it expected?

Reply:
Yes, the dashboard admin can not access the the global data source in the workspace